src/modules/kvs/lookup.h

```diff
@@ -36,6 +36,12 @@ bool lookup_validate (lookup_t *lh);
  * an error occurred or not */
 int lookup_get_errnum (lookup_t *lh);
 
+/* if user wishes to stall, but needs future knowledge to fail and
```

Comment:
is that a real use case?   is there ever a reason to stall the request if you've hit an error?  Seems like you would want to immediately abort and get the response to the user.

Reply:
So the way I've handled it in the past (with the commit API) was that if multiple RPCs are sent successfully then one fails, I wait for the in-flight rpcs to complete then return the error to the user.  So these get/set aux errnums are the "flag" for the callback function to return an error to the user.  See ```commit_apply()``` function in kvs.c and you'll see what I do.

This perhaps could be dealt with an alternate way.  Perhaps an error could be returned to the user immediately and get/set aux errnum could be a flag informing the callback function "this rpc already errored, don't send user a response".  It'd simply be a logic change.  Perhaps for another issue though?

Reply:
Ah, sorry, I think that's fine for now.  It's an error case, so trading a little latency for simplicity is probably a wise move.  If you feel that change could actually simplify the code here and in the commit API, then I'd suggest opening a bug for later.

Reply:
I think code wise it's not too much different each way.  Of course returning the error to the user earlier shortens the latency a bit.  I'll create an issue so we don't forget about this.

Reply:
Actually, thinking about it now, the code logic would probably be more confusing.  Because I'd have to keep some auxiliary-data structures around longer than they are currently used.

For example in ```commit_apply()```, after we reply to the user we call ```commit_mgr_remove_commit()``` to remove/destroy a commit context.  If I were to change the logic to return an error to the user immediately, I can't destroy this.   It still needs to exist for the rpcs that haven't finished.

Think it'd be better to keep it the way it is right now.